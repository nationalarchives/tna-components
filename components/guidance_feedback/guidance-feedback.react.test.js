import React from 'react';
import GuidanceFeedback from './guidance-feedback.react';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

test('test', () => {
  expect(true).toBe(true);
});
// // Snapshot 1
test('Show the relevant form elements when click on NO or YES button', () => {
  const component = renderer.create(
    <GuidanceFeedback />
  );

  let submitEvent = { 
        preventDefault: () => {}
  };

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
  // Click on NO button
  tree.children[2].props.onClick(submitEvent);

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Click on YES button
  tree.children[3].props.onClick(submitEvent);

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});

// Snapshot 2
test('Hide the form and button elements on submit', () => {
  const component = renderer.create(
    <GuidanceFeedback />
  );

  let submitEvent = { 
    preventDefault: () => {} 
  };

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.children[4].children[1].props.onClick(submitEvent);

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});

// Snaphost 3
test('Revert the widget back to the initial state when click on CANCEL', () => {
  const component = renderer.create(
    <GuidanceFeedback />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Click on CANCEL button
  tree.children[4].children[2].props.onClick();

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});

// Unit testing using Enzyme
test('Check for the absence of the Guidance Feedback component', () => {
  const component = shallow(
    <GuidanceFeedback />
  );
  expect(component.exists()).toBeTruthy();
});

// Unit testing
test('Test the methods', () => {
  const component = mount(
    <GuidanceFeedback />
  );

  const obj = {};

  component.instance().buildGTMObj(obj);

  expect(typeof component.instance().buildGTMObj(obj) === 'object').toBeTruthy();

  expect(component.instance().buildGTMObj(obj)).toHaveProperty('event');
  expect(component.instance().buildGTMObj(obj)).toHaveProperty('eventAction');
  expect(component.instance().buildGTMObj(obj)).toHaveProperty('eventCategory');
  expect(component.instance().buildGTMObj(obj)).toHaveProperty('eventLabel');

  component.instance().pushInDataLayer(obj);

  expect(typeof component.instance().pushInDataLayer(obj) === 'object').toBeTruthy();

});