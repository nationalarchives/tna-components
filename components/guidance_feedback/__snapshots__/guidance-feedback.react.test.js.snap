// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hide the form and button elements on submit 1`] = `
<form
  action=""
  className="react-component-gf"
  id="guidance-feedback"
>
  <h2>
    Feedback
  </h2>
  <h3
    id="aria"
  >
    Did you find the guidance you needed?
  </h3>
  <button
    aria-describedby="aria"
    className="btn--no show"
    name="btn--no"
    onClick={[Function]}
    type="button"
  >
    No
  </button>
  <button
    aria-describedby="aria"
    className="btn--yes show"
    name="btn--yes"
    onClick={[Function]}
    type="button"
  >
    Yes
  </button>
  <fieldset
    className="hide"
  >
    <div>
      <label
        htmlFor=""
      >
        
      </label>
      <textarea
        id=""
        name=""
      />
    </div>
    <input
      className="btn btn--send"
      onClick={[Function]}
      type="submit"
      value="Send"
    />
    <input
      className="btn btn--cancel"
      type="reset"
      value="Cancel"
    />
  </fieldset>
</form>
`;

exports[`Hide the form and button elements on submit 2`] = `
<form
  action=""
  className="react-component-gf"
  id="guidance-feedback"
>
  <h2>
    Feedback
  </h2>
  <h3
    id="aria"
  >
    Thank you for your feedback.
  </h3>
  <button
    aria-describedby="aria"
    className="btn--no hide"
    name="btn--no"
    onClick={[Function]}
    type="button"
  >
    No
  </button>
  <button
    aria-describedby="aria"
    className="btn--yes hide"
    name="btn--yes"
    onClick={[Function]}
    type="button"
  >
    Yes
  </button>
  <fieldset
    className="hide"
  >
    <div>
      <label
        htmlFor=""
      >
        
      </label>
      <textarea
        id=""
        name=""
      />
    </div>
    <input
      className="btn btn--send"
      onClick={[Function]}
      type="submit"
      value="Send"
    />
    <input
      className="btn btn--cancel"
      type="reset"
      value="Cancel"
    />
  </fieldset>
</form>
`;

exports[`Show the relevant form elements when click on button NO or YES 1`] = `
<form
  action=""
  className="react-component-gf"
  id="guidance-feedback"
>
  <h2>
    Feedback
  </h2>
  <h3
    id="aria"
  >
    Did you find the guidance you needed?
  </h3>
  <button
    aria-describedby="aria"
    className="btn--no show"
    name="btn--no"
    onClick={[Function]}
    type="button"
  >
    No
  </button>
  <button
    aria-describedby="aria"
    className="btn--yes show"
    name="btn--yes"
    onClick={[Function]}
    type="button"
  >
    Yes
  </button>
  <fieldset
    className="hide"
  >
    <div>
      <label
        htmlFor=""
      >
        
      </label>
      <textarea
        id=""
        name=""
      />
    </div>
    <input
      className="btn btn--send"
      onClick={[Function]}
      type="submit"
      value="Send"
    />
    <input
      className="btn btn--cancel"
      type="reset"
      value="Cancel"
    />
  </fieldset>
</form>
`;

exports[`Show the relevant form elements when click on button NO or YES 2`] = `
<form
  action=""
  className="react-component-gf"
  id="guidance-feedback"
>
  <h2>
    Feedback
  </h2>
  <h3
    id="aria"
  >
    Did you find the guidance you needed?
  </h3>
  <button
    aria-describedby="aria"
    className="btn--no hide"
    name="btn--no"
    onClick={[Function]}
    type="button"
  >
    No
  </button>
  <button
    aria-describedby="aria"
    className="btn--yes hide"
    name="btn--yes"
    onClick={[Function]}
    type="button"
  >
    Yes
  </button>
  <fieldset
    className="show"
  >
    <div>
      <label
        htmlFor="field-no"
      >
        What did you expect to find?
      </label>
      <textarea
        id="field-no"
        name="field-no"
      />
    </div>
    <input
      className="btn btn--send"
      onClick={[Function]}
      type="submit"
      value="Send"
    />
    <input
      className="btn btn--cancel"
      type="reset"
      value="Cancel"
    />
  </fieldset>
</form>
`;

exports[`Show the relevant form elements when click on button NO or YES 3`] = `
<form
  action=""
  className="react-component-gf"
  id="guidance-feedback"
>
  <h2>
    Feedback
  </h2>
  <h3
    id="aria"
  >
    Did you find the guidance you needed?
  </h3>
  <button
    aria-describedby="aria"
    className="btn--no show"
    name="btn--no"
    onClick={[Function]}
    type="button"
  >
    No
  </button>
  <button
    aria-describedby="aria"
    className="btn--yes show"
    name="btn--yes"
    onClick={[Function]}
    type="button"
  >
    Yes
  </button>
  <fieldset
    className="hide"
  >
    <div>
      <label
        htmlFor="field-yes"
      >
        Any comments on your experience?
      </label>
      <textarea
        id="field-yes"
        name="field-yes"
      />
    </div>
    <input
      className="btn btn--send"
      onClick={[Function]}
      type="submit"
      value="Send"
    />
    <input
      className="btn btn--cancel"
      type="reset"
      value="Cancel"
    />
  </fieldset>
</form>
`;
